controllers/profiles/followAnonUser.js

```diff
@@ -0,0 +1,105 @@
+const moment = require('moment');
+
+const UserFollowUserFunction = require('../../databases/functions/userFollowUser');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const SuccessResponse = require('../../utils/response/SuccessResponse');
+
+const {UserFollowUser, UserFollowUserHistory, User, sequelize} = require('../../databases/models');
+const Getstream = require('../../vendor/getstream');
+const {addForFollowUser} = require('../../services/score');
+const BetterSocialCore = require('../../services/bettersocial');
+const UsersFunction = require('../../databases/functions/users');
+const {sendFollowMainFeedF2} = require('../../services/queue/mainFeedF2');
+
+module.exports = async (req, res) => {
+  const {user_id_followed, follow_source} = req.body;
+  const {user} = req;
+
+  if (req?.userId === user_id_followed)
+    return ErrorResponse.e403(res, 'Only allow following other profiles');
+
+  if (
+    await UserFollowUserFunction.checkIsUserFollowing(UserFollowUser, req?.userId, user_id_followed)
+  )
+    return ErrorResponse.e409(res, 'You have followed this user');
+
+  const targetUser = await UsersFunction.findUserById(User, user_id_followed);
+  if (!targetUser) {
+    return ErrorResponse.e404(res, 'User not found');
+  } else if (!targetUser.is_anonymous) {
+    return ErrorResponse.e403(res, 'Should be anonymous user');
+  }
+
+  try {
+    await sequelize.transaction(async (t) => {
+      const userFollowUser = await UserFollowUserFunction.registerAddFollowUser(
+        UserFollowUser,
+        UserFollowUserHistory,
+        req?.userId,
+        [user_id_followed],
+        follow_source,
+        t
+      );
+      return userFollowUser;
+    });
+  } catch (e) {
+    console.log('Error in follow anonymos user sql transaction');
+    return ErrorResponse.e409(res, e.message);
+  }
+
+  try {
+    const followUserExclusive = Getstream.feed.followUserExclusive(req?.userId, user_id_followed);
+    const followUser = Getstream.feed.followUser(req?.token, req?.userId, user_id_followed);
+    const followMainFeedFollowing = Getstream.feed.followMainFeedFollowing(
```

Comment:
These needs to be changed.
For follow anon, the feed follows should be like this
1. main_feed:signed-self-id follows user_anon:target-anon-user-id
2. main_feed_following:signed-self-id follows user_anon:target-anon-user-id

Reply:
fixed here : https://github.com/BetterSocial/user-api/pull/585